File: http-led.py
try:
  import usocket as socket
except:
  import socket

from machine import Pin
import network

import esp
esp.osdebug(None)

import gc
gc.collect()

ssid = 'COLOCAR_SSID'
senha = 'COLOCAR_SENHA'
estacao = network.WLAN(network.STA_IF)
estacao.active(True)
estacao.connect(ssid, senha)

while estacao.isconnected() == False:
  pass

print('Conexao realizada.')
print(estacao.ifconfig())

led = Pin(16, Pin.OUT) # D0 (GPIO16)

def pagina_web(arquivo):
  conteudo = ''
  a = open(arquivo)
  conteudo = a.read()
  a.close()
  return conteudo

s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.bind(('', 80))
s.listen(5)

while True:
  conexao, endereco = s.accept()
  print('Conexao de %s' % str(endereco))
  requisicao = conexao.recv(1024)
  requisicao = str(requisicao)
  print('Conteudo = %s' % requisicao)
  if requisicao.find('/led/1') != -1:
    print('Acender')
    led.value(1)
  elif requisicao.find('/led/0') != -1:
    print('Apagar')
    led.value(0)
  else:
    led.value(0)
  html = pagina_web('http-led-bootstrap.html')
  conexao.send(html)
  conexao.close()

File: http-led.lua
wifi.setmode(wifi.STATION)
station_cfg={}
station_cfg.ssid="COLOCAR_SSID"
station_cfg.pwd="COLOCAR_SENHA"
station_cfg.save=false
wifi.sta.config(station_cfg)
wifi.sta.connect()
print("IP do Servidor: ")
print ("Espere...")
tmr.alarm(0, 1000, 1, function()
  if wifi.sta.getip() ~= nil then
    print("Pronto! IP do Servidor: " .. wifi.sta.getip())
    tmr.stop(0)
  end
end)
srv=net.createServer(net.TCP) 
srv:listen(80, function(conn) 
  conn:on("receive", function(conn, requisicao) 
    pino=0 -- D0 (GPIO16)
    gpio.mode(pino, gpio.OUTPUT)  
    --print(requisicao)
    if string.find(requisicao, " /led/1") ~= nil then
      print("Acender")
      gpio.write(pino, gpio.HIGH)
    else 
      if string.find(requisicao, " /led/0") ~= nil then
        print("Apagar")
        gpio.write(pino, gpio.LOW)
      else
        gpio.write(pino, gpio.LOW)
      end
    end
    conteudo = ""
    file.open("http-led-bootstrap.html", "r")
    while true do
      linha = file.readline()
      if linha == nil then
        break
      end
      conteudo = conteudo .. linha
    end
    file.close()
    --print(conteudo)
    conn:send(conteudo)
  end)
  conn:on("sent", function(conn) conn:close() end) 
end)
